Userland/Applications/Piano/MainWidget.cpp

```diff
@@ -19,53 +19,65 @@
 #include <LibGUI/Menu.h>
 #include <LibGUI/TabWidget.h>
 
+ErrorOr<NonnullRefPtr<MainWidget>> MainWidget::create(TrackManager& manager, AudioPlayerLoop& loop)
+{
+    auto widget = TRY(adopt_nonnull_ref_or_enomem(new (nothrow) MainWidget(manager, loop)));
+    widget->m_player_widget = TRY(PlayerWidget::create(widget->m_track_manager, widget->m_audio_loop));
+    widget->initialize();
+    return widget;
+}
+
 MainWidget::MainWidget(TrackManager& track_manager, AudioPlayerLoop& loop)
     : m_track_manager(track_manager)
     , m_audio_loop(loop)
+{
+}
+
+void MainWidget::initialize()
 {
     set_layout<GUI::VerticalBoxLayout>();
     layout()->set_spacing(2);
     layout()->set_margins(2);
     set_fill_with_background_color(true);
 
-    m_wave_widget = add<WaveWidget>(track_manager);
+    m_wave_widget = add<WaveWidget>(m_track_manager);
     m_wave_widget->set_fixed_height(100);
 
     m_tab_widget = add<GUI::TabWidget>();
-    m_roll_widget = m_tab_widget->add_tab<RollWidget>("Piano Roll", track_manager);
+    m_roll_widget = m_tab_widget->add_tab<RollWidget>("Piano Roll", m_track_manager);
```

Comment:
You can use `TRY(try_add_tab())` instead, which will propagate the error when creating the RollWidget.